### PR TITLE
[ONNX] Add support for negative calculated padding

### PR DESCRIPTION
**Pull request by tsocha**

### CODE REVIEW

File: src/ngraph/frontend/onnx_import/op/conv_transpose.cpp [hunk 1]
```diff
@@ -171,23 +171,27 @@ namespace ngraph
...
-                            if (static_cast<int>(padding_below[i]) < 0)
+                            padding_below[i] = strides[i] * (data_shape[i + 2] - 1) +
+                                               dilations[i] * (weights_shape[i + 2] - 1) -
+                                               data_dilation_strides[i] *
+                                                   (output_shape[i] - output_padding[i] - 1);
+                            if (padding_below[i] < 0)
                             {
-                                output_padding[i] = -static_cast<int>(padding_below[i]);
-                                padding_below[i] = 0;
+                                // (int) -9 / 2 = -5 but we need -4
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/op/conv_transpose.cpp]: Add comment: `Divide by 2 and round both positive and negative values to nearest integer`

File: src/ngraph/frontend/onnx_import/op/conv_transpose.cpp [hunk 2]
```diff
@@ -171,23 +171,27 @@ namespace ngraph
...
-                                                dilations[i] * (weights_shape[i + 2] - 1) -
-                                                data_dilation_strides[i] *
-                                                    (output_shape[i] - output_padding[i] - 1)) /
-                                               2;
-                            if (static_cast<int>(padding_below[i]) < 0)
+                            padding_below[i] = strides[i] * (data_shape[i + 2] - 1) +
+                                               dilations[i] * (weights_shape[i + 2] - 1) -
+                                               data_dilation_strides[i] *
+                                                   (output_shape[i] - output_padding[i] - 1);
+                            if (padding_below[i] < 0)
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/op/conv_transpose.cpp]: Do you need this condition?

File: src/ngraph/frontend/onnx_import/op/conv_transpose.cpp [hunk 3]
```diff
@@ -171,23 +171,27 @@ namespace ngraph
...
                             {
-                                output_padding[i] = -static_cast<int>(padding_below[i]);
-                                padding_below[i] = 0;
+                                // (int) -9 / 2 = -5 but we need -4
+                                // (int) -9 --> 9 / 2 = 4 --> -4
+                                padding_below[i] = -(-padding_below[i] / 2);
+                            }
+                            else
+                            {
+                                padding_below[i] /= 2;
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/op/conv_transpose.cpp]: Add a comment why we need to divide this value by 2

File: src/ngraph/frontend/onnx_import/op/conv_transpose.cpp (hunk 1 above)
Comment (tsocha) [reply src/ngraph/frontend/onnx_import/op/conv_transpose.cpp]: We do not rounding values but cutting the fractional part.

File: src/ngraph/frontend/onnx_import/op/conv_transpose.cpp (hunk 3 above)
Comment (tsocha) [reply src/ngraph/frontend/onnx_import/op/conv_transpose.cpp]: to get symmetric above and below padding.

File: src/ngraph/frontend/onnx_import/op/conv_transpose.cpp (hunk 2 above)
Comment (tsocha) [reply src/ngraph/frontend/onnx_import/op/conv_transpose.cpp]: yes.

**Pull request merged**